src/main/host/host.rs

```diff
@@ -0,0 +1,29 @@
+use crate::cshadow;
+
+pub struct Host {
+    chost: *mut cshadow::Host,
+}
+
+impl Host {
+    /// For now, this should only be called via Worker, to borrow the current
+    /// Host. This will ensure there is only one reference to a given Host
+    /// in Rust.
+    ///
+    /// SAFETY: `p` must point to a valid c::Host, to which this Host will
+    /// have exclusive access over its lifetime. `p` must outlive the returned object.
+    pub unsafe fn borrow_from_c(p: *mut cshadow::Host) -> Self {
+        assert!(!p.is_null());
+        Host { chost: p }
+    }
+
+    pub fn name(&self) -> &str {
+        let slice = unsafe { std::ffi::CStr::from_ptr(cshadow::host_getName(self.chost)) };
```

Comment:
I don't think it actually can be NULL, can it? You can't actually define a host in the config without giving it a name...?

Added an assertion to `host_new` to be sure.

Reply:
Within Shadow I don't think it is ever NULL, I was just thinking about within the scope of the `Host` type specifically since its constructor accepts a NULL value.